module.c

```diff
@@ -442,6 +503,77 @@ static PyObject *context_gc(ContextData *self) {
 	Py_RETURN_NONE;
 }
 
+static JSValue js_c_function(
+    JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv, int magic) {
+	ContextData *self = (ContextData *)JS_GetContextOpaque(ctx);
+	PythonCallableNode *node = self->python_callables;
+	while (node && node->magic != magic) {
+		node = node->next;
+	}
+	if (!node) {
+		return JS_ThrowInternalError(self->context, "Internal error.");
+	}
+
+	PyObject *args = PyTuple_New(argc);
+	if (!args) {
+		return JS_ThrowOutOfMemory(self->context);
+	}
+	for (int i = 0; i < argc; ++i) {
+		PyObject *arg = quickjs_to_python(self, JS_DupValue(self->context, argv[i]));
```

Comment:
You do not need to use the `JS_DupValue`, if direct return the resoult of `quickjs_to_python` without call `JS_FreeValue` in case `if (tag == JS_TAG_OBJECT || tag == JS_TAG_MODULE)` .

Reply:
Well, `quickjs_to_python` could have been implemented either way I suppose. But now it consumes its argument and is documented that way, so we need to do this here as well.

Reply:
I think we can change the document, add e.g. "refcount reduced by 1, if need". In fact, in case `if (tag == JS_TAG_OBJECT || tag == JS_TAG_MODULE)`, its refcount should not be reduced, the return_value referrered to it.

Reply:
Feels a little out of scope for this PR though?

Reply:
Yes, you are right. But I see you commented this in unittest, maybe we can resolve it in this PR, or resolve in new one?